### ISSUE TITLE
Dumping fails when one stack consumes the output of another stack

### ISSUE THREAD

**Issue by jonathanunderwood** [remind101/stacker#256]
With 1.0.0a1 dumping doesn't work:

```
$ stacker build -vv -d  dump -r eu-west-1 ./conf/capone-formisimo-processor.env ./conf/ec2-instance.yaml
[2016-10-29T16:30:48] INFO stacker.commands.stacker:33(configure): Using Default AWS Provider
Traceback (most recent call last):
  File "/usr/local/bin/stacker", line 9, in <module>
    args.run(args)
  File "/usr/local/lib/python2.7/site-packages/stacker/commands/stacker/build.py", line 52, in run
    dump=options.dump)
  File "/usr/local/lib/python2.7/site-packages/stacker/actions/base.py", line 125, in execute
    self.run(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/stacker/actions/build.py", line 285, in run
    plan.dump(dump)
TypeError: dump() takes exactly 3 arguments (2 given)
```

It looks like the directory option isn't propagated through to Plan.dump.

**Comment by jonathanunderwood** [comment]
With the latest 1.0.0a3 release:

```
$ stacker build -vv -d  dump -r eu-west-1 cloudformation/conf/capone-formisimo-processor.env cloudformation/conf/ec2-instance.yaml
[2016-11-04T17:11:33] INFO stacker.commands.stacker:33(configure): Using Default AWS Provider
[2016-11-04T17:11:33] INFO stacker.plan:314(dump): Dumping "Create/Update stacks"...
[2016-11-04T17:11:33] DEBUG stacker.blueprints.base:295(setup_parameters): No parameters defined.
[2016-11-04T17:11:33] INFO stacker.plan:328(dump): Writing stack "capone-formisimo-processor-mongodb-ebs-volumes" -> dump/mongodb-ebs-volumes-48c6cbc4.json
Traceback (most recent call last):
  File "/usr/local/bin/stacker", line 9, in <module>
    args.run(args)
  File "/usr/local/lib/python2.7/site-packages/stacker/commands/stacker/build.py", line 52, in run
    dump=options.dump)
  File "/usr/local/lib/python2.7/site-packages/stacker/actions/base.py", line 125, in execute
    self.run(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/stacker/actions/build.py", line 288, in run
    plan.dump(directory=dump, context=self.context)
  File "/usr/local/lib/python2.7/site-packages/stacker/plan.py", line 323, in dump
    provider=None,
  File "/usr/local/lib/python2.7/site-packages/stacker/stack.py", line 155, in resolve
    resolve_variables(self.variables, context, provider)
  File "/usr/local/lib/python2.7/site-packages/stacker/variables.py", line 74, in resolve_variables
    variable.resolve(context, provider)
  File "/usr/local/lib/python2.7/site-packages/stacker/variables.py", line 134, in resolve
    resolved_lookups = resolve_lookups(self.lookups, context, provider)
  File "/usr/local/lib/python2.7/site-packages/stacker/lookups/registry.py", line 63, in resolve_lookups
    provider=provider,
  File "/usr/local/lib/python2.7/site-packages/stacker/lookups/handlers/output.py", line 26, in handler
    raise ValueError('Provider is required')
ValueError: Provider is required

```

**Comment by jonathanunderwood** [comment]
OK, the above happens when there is one stack using the output of another stack. I don't have a good idea as to how that could be fixed, since when dumping, the first stack may not exist, so it's not possible to pull variables from AWS for those outputs. Perhaps it would be better to put placeholders in the second stack to at least make dump functional as a debugging helper.

**Comment by phobologic** [comment]
@jonathanunderwood I believe that the right way to deal with this is to use the `--vars` flag to replace the variables that usually call `${output}`.

it might be possible for us to ignore the output lookups when doing a dump, let me look into that.

**Comment by phobologic** [comment]
Hmm - we could, in theory, provide default values for each of the types, but that might get messy.  @jonathanunderwood any idea what you would expect for this to do when you're doing cross stack output lookups?

**Comment by jonathanunderwood** [comment]
Good question. I tend to be using this for debugging purposes, so if the parameter was tokenized to read something like {{${otherstack:var}}} I'd be happy. That would obviously be an invalid cloud formation template, but at least it would be usable for debugging purposes.

**Comment by phobologic** [comment]
Hey @jonathanunderwood - I believe this should have been mostly fixed by #410.  I'm going to go ahead and close this, but feel free to re-open if that doesn't fix it for you.  Thanks!

**Comment by jonathanunderwood** [comment]
Great, thanks Michael. Will give it a test when I have chance.
